pygtrie.py

```diff
@@ -1364,13 +1364,35 @@ def add(self, key):
             self._trie[key:] = True
 
     def discard(self, key):
-        raise NotImplementedError(
-            'Removing keys from PrefixSet is not implemented.')
+        try:
+            node, _ = self._get_node(key)
+        except:
+            pass
+
+        if node and node.children:
+            for child in node.children:
+                self.discard(child)
+        else:
+            node.children = {}
+            node.value = _SENTINEL
+        
+
 
+    # Raises KeyError if elem is not contained in the set.
     def remove(self, key):
-        raise NotImplementedError(
-            'Removing keys from PrefixSet is not implemented.')
+        try:
+            node, _ = self._get_node(key)
+        except KeyError:
+            return 0
```

Comment:
Uh? Why are you catching the exception? It should be propagated.

More importantly, why are you duplicating the implementation? It’s much better to implement ``discard`` in terms of ``remove``, i.e.:

```
    def remove(self, key):
        …

    def discard(self, key):
        try:
            self.remove(key)
        except KeyError:
            pass
```